syntax = "proto3";

package serializator;

import "map_renderer.proto";
import "graph.proto";

message Coordinates {
    double latitude = 1;
    double longitude = 2;
}

message BusStop {
    string name = 1;
    Coordinates coordinates = 2;
}

message BusRoute {
    string number = 1;
    repeated BusStop stops = 2;
    bool circle = 3;
}

message StopDistance {
  string from_stop = 1;
  string to_stop = 2;
  double distance = 3;
}

message Base {
  repeated BusRoute buses = 1;
  repeated BusStop stops = 2;
  repeated StopDistance stop_distance_map = 3;
  RenderSettings map_settings = 4;
  RouteSettings route_settings = 5;
}